docs/config.md

```diff
@@ -0,0 +1,94 @@
+# Config file options
+
+SOCI has a multitude of variables that can customize SOCI behavior to best suit users' needs. This doc will break down the possible values for each variable, where the variable can be found, and what it can do.
+
+This doc assumes the usage of a TOML file. Per TOML specs, Anything in [brackets] is the required header for the set of variables.
+
+An example configuration file can be found in config/config.toml.
+
+The following format will be used throughout this doc:
+
+## File path
+### [toml_header]
+- `toml_var_name` (type) — Brief description. Default: some value.
+
+#
+
+# Uncategorized
+
+This set of variables must be at the top of your TOML file due to not belonging to any TOML header.
+
+## config/fs.go
+### FSConfig
+- `resolve_result_entry` (int) — Max amount of entries allowed in the cache. Default: 30.
+- `debug` (bool) — Enables debugging for go-fuse in logs. This often emits sensitive data, so this should be false in production. Default: false.
+- `disable_verification` (bool) — Allows skipping TOC validation, which can give slight performance improvements if files have already been verified elsewhere. Default: false.
+- `no_prometheus` (bool) — Toggle prometheus metrics. Default: false.
+- `mount_timeout_sec` (int) — Timeout for mount if a layer can't be resolved. Default: 30.
+- `fuse_metrics_emit_wait_duration_sec` (int) — Time until snapshotter emits FUSE metrics in logs. Default: 60.
+
+## config/config.go
+### Config
+- `metrics_address` (string) — If empty, no metrics will be polled. Default: "".
+- `metrics_network` (string) — Chooses protocol to send metrics over (e.g. tcp, unix, etc). Default: "tcp".
+- `no_prometheus` — Defined [above](#configfsgofsconfig), cannot be redeclared.
+- `debug_address` (string) — Address where [go pprof](https://pkg.go.dev/net/http/pprof) server will listen. If empty, no logs will be emitted. Default: "".
+- `metadata_store` (string) — Metadata storage type. Only "db" is valid. Default: "db".
+
+# Categorized
+
+## config/fs.go
+
+### [http]
+- `MaxRetries` (int) — Max retries before giving up on a network request. Default: 8.
+- `MinWaitMsec` (int) — Min time between network request attempts. Default: 30.
+- `MaxWaitMsec` (int) — Max time between network request attempts. Default: 30.
+- `DialTimeoutMsec` (int) — Max time for a connection before timeout. Default: 3000.
+- `ResponseHeaderTimeoutMsec` (int) — Maximum duration waiting for response headers before timeout. Default: 3000.
+- `RequestTimeoutMsec` (int) — Maximum duration waiting for entire request before timeout. Default: 30000.
+
+### [blob]
```

Comment:
Would it be helpful to explain exactly what a blob is? Feels like it would be difficult to elaborate on what they do without explaining some of the technical detail, but again I wouldn't want this to be too overwhelming.